tests/xfstests/run.pm

```diff
@@ -170,6 +176,45 @@ sub tests_from_category {
     return @tests;
 }
 
+# Return matched exclude tests from groups in @GROUPLIST
+# return structure - hash
+# Group name start with ! will exclude in test, and expected to use to update blacklist
+sub exclude_grouplist {
+    my %tests_list = ();
+    foreach my $group_name (@GROUPLIST) {
+        next if ($group_name !~ /^\!/);
+        $group_name = substr($group_name, 1);
+        my $cmd = "cat $INST_DIR/tests/$FSTYPE/group | grep $group_name | awk '{printf \"$FSTYPE/\"}{printf \$1}{printf \",\"}' > tmp.group";
```

Comment:
you can combine cat | grep | awk in one awk, https://blog.jpalardy.com/posts/skip-grep-use-awk/

Reply:
@czerw ok, good to know. Fixed.

Reply:
new VR http://10.67.133.102/tests/1171